activesupport/test/option_merger_test.rb

```diff
@@ -114,6 +114,17 @@ def test_option_merger_implicit_receiver
     assert_equal expected, @options
   end
 
+  def test_with_options_hash_like
+    hash_like = Struct.new :hash do
```

Comment:
Might be best to avoid `Struct` altogether as it defines a lot of hash like methods.

A regular object would do.